ecommerce/courses/utils.py

```
@@ -25,29 +25,75 @@ def mode_for_product(product):
     return mode
 
 
-def get_course_info_from_catalog(site, product):
-    """ Get course or course_run information from Discovery Service and cache """
-    if product.is_course_entitlement_product:
-        key = product.attr.UUID
-    else:
-        key = CourseKey.from_string(product.attr.course_key)
+def _get_discovery_response(site, cache_key, resource, resource_id=None):
+    """
+    Return the discovery endpoint result of given resource or cached response if its already been cached.
 
-    api = site.siteconfiguration.discovery_api_client
-    partner_short_code = site.siteconfiguration.partner.short_code
+    Arguments:
+        site (Site): Site object containing Site Configuration data
+        cache_key (str): Cache key for given resource
+        resource_id (int or str): Identifies a specific resource to be retrieved
 
-    cache_key = u'courses_api_detail_{}{}'.format(key, partner_short_code)
-    cache_key = hashlib.md5(cache_key.encode('utf-8')).hexdigest()
+    Returns:
+        dict: resource's information for given resource_id received from Discovery API
+    """
+    api = site.siteconfiguration.discovery_api_client
     course_cached_response = TieredCache.get_cached_response(cache_key)
     if course_cached_response.is_found:
         return course_cached_response.value
 
-    if product.is_course_entitlement_product:
-        course = api.courses(key).get()
+    endpoint = getattr(api, resource)
+    if resource == 'course_runs':
+        response = endpoint(resource_id).get(partner=site.siteconfiguration.partner.short_code)
     else:
```

can we add add `elif` here to confirm resource is `course` instead of just assuming it would be a `course`?
and in else part we can raise an exception that resource is INVALID.
this will also increase readability here.

---

in else condition, there can be `course` and `catalogs` as a resource. We needed if condition because we are further filtering out on partner shortcode in case of `course_run`